product_variant_sale_price/views/product_view.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+
+    <record id="product_template_form_view" model="ir.ui.view">
+        <field name="name">product.template.form.inherited</field>
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_form_view"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='list_price']" position="attributes">
+                <attribute name="attrs">{'invisible': [
+                                            ('product_variant_count', '>', 1)
+                                        ]}</attribute>
+            </xpath>
+        </field>
+    </record>
+
+    <record id="product_normal_form_view_default_check" model="ir.ui.view">
+       <field name="model">product.product</field>
+       <field name="inherit_id" ref="product.product_normal_form_view" />
+       <field name="arch" type="xml">
+            <xpath expr="//field[@name='uom_id']" position="after">
+                <field name="fix_price"
+                       attrs="{'invisible': [('product_variant_count', '=', 1)]}"/>
+            </xpath>
+       </field>
+    </record>
+
```

Comment:
You can improve functionality including this view:


    <record id="product_variant_easy_edit_price_view" model="ir.ui.view">
       <field name="model">product.product</field>
       <field name="inherit_id" ref="product.product_variant_easy_edit_view" />
       <field name="arch" type="xml">
            <xpath expr="//field[@name='lst_price']" position="after">
                <field name="fix_price"
                       attrs="{'invisible': [('product_variant_count', '=', 1)]}"/>
            </xpath>
       </field>
    </record>

Reply:
Yes, I'm going to do it. :smile: